src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -198,6 +199,25 @@ export class RoslynLanguageServer {
             );
         });
 
+        this._languageClient.onNotification(RoslynProtocol.ProjectLoadFailureNotification.type, async (failure) => {
+            const showLogs = vscode.l10n.t('Show C# logs');
+            const isError = failure.messageType === MessageType.Error;
+            const projectName = path.basename(failure.projectPath);
+            const title = isError
+                ? vscode.l10n.t('There were errors loading project {0}. See logs for details.', projectName)
+                : vscode.l10n.t('There were warnings loading project {0}. See logs for details.', projectName);
+            const options: vscode.MessageOptions = {
+                modal: false,
+                detail: failure.message,
+            };
+            const selection = isError
+                ? await vscode.window.showErrorMessage(title, options, showLogs)
+                : await vscode.window.showWarningMessage(title, options, showLogs);
+            if (selection === showLogs) {
+                _channel.show();
+            }
+        });
```

Comment:
Do we just want to make this be a more general endpoint for any sort of failure, not just project loads?